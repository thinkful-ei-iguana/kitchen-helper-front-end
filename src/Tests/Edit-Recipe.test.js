import React from 'react';
import ReactDOM from 'react-dom';
import EditRecipe from '../Components/Edit-Recipe';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
        <EditRecipe />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer.create(<BrowserRouter>
        <EditRecipe />
    </BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});