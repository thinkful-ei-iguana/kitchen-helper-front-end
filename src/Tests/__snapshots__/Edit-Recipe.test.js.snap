// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="EditRecipe"
>
  <header
    className="Creation-Header"
  />
  <form
    className="Creation-Form"
    onChange={[Function]}
    onSubmit={[Function]}
  >
    <label
      className="field a-field a-field_a2"
    >
      <input
        className="field__input a-field__input"
        name="title"
        placeholder="Title"
        required={true}
      />
      <span
        className="a-field__label-wrap"
      >
        <span
          className="a-field__label"
        />
      </span>
    </label>
    <label
      className="field a-field a-field_a2"
    >
      <input
        className="field__input a-field__input"
        name="recipe_description"
        placeholder="Description"
        required={true}
        type="textfield"
      />
      <span
        className="a-field__label-wrap"
      >
        <span
          className="a-field__label"
        />
      </span>
    </label>
    <label>
      <input
        className="field__input a-field__input"
        name="recipe_ingredients"
        placeholder="Ingredients"
        required={true}
        type="text"
      />
      <span
        className="a-field__label"
      />
    </label>
    <label>
      <input
        className="field__input a-field__input"
        name="time_to_make"
        placeholder="Time to make it?"
        required={true}
        type="text"
      />
      <span
        className="a-field__label"
      />
    </label>
    <div
      className="btn-row"
    >
      <button
        className="submitRecipeEdit"
      >
        Submit
      </button>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="cancelEditRecipe"
        >
          Cancel
        </button>
      </a>
    </div>
  </form>
</div>
`;
